app/views/posts/_new_form.html.erb

```diff
@@ -0,0 +1,5 @@
+<%= form_with model: @post, url: user_posts_path(@post.author) do |form| %>
+    <%= form.text_field :title, placeholder: "Title" %>
+    <%= form.text_area :text, placeholder: "Write something..."%>
+    <%= form.submit "Create a post", class: "submit-button" %>
+<% end %>
```

Comment:
- [ **OPTIONAL** ] I think it would be better if you add buttons to add a new **Post** on both `app/views/users/show.html.erb` and `app/views/posts/index.html.erb` pages. Right now to create a Post, users are supposed to manually write the appropriate route, which is not best practice for someone who doesn't have skills in programming.